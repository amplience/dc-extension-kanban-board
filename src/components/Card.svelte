<script>
  export let title: string = '';
  export let subtitle: string = '';
  export let footer: string = '';
</script>

<style lang="scss">
  .card {
    box-sizing: border-box;
    background-color: white;
    padding: 10px 10px 20px 10px;
    border-radius: 3px;
    margin: 10px;

    &:focus {
      outline-color: #039be5;
    }
  }

  .title {
    color: #333;
    font-size: 13px;
    margin-bottom: 4px;
  }
  .subtitle,
  .footer {
    font-size: 0.9em;
    color: #999;
    font-size: 13px;
  }
</style>

<section class="card">
  <h1 class="title">{title}</h1>
  <span class="subtitle">{subtitle}</span>
  <footer class="footer">{footer}</footer>
</section>
