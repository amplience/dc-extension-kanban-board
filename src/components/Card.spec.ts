import { render } from '@testing-library/svelte';
import Card from './Card.svelte';

describe('card component', () => {
  it('should render a card component', () => {
    const client = {
      dashboardSdk: {
        applicationNavigator: {
          openContentItem: jest.fn(),
        },
      },
    };

    const { container } = render(Card, {
      contentItem: {
        label: 'LABEL',
        schema: 'schema',
        lastModifiedDate: '2020-01-01T00:00:59Z',
      },
      client: client,
      contentTypeLookup: {
        schema: {
          settings: {
            label: 'SCHEMA LABEL',
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
