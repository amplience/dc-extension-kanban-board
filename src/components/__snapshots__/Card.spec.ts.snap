// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card component should render a card component 1`] = `
<div>
  <section
    class="card svelte-1rse9oa"
  >
    <a
      class="svelte-1rse9oa"
      target="_top"
    >
      <h1
        class="title svelte-1rse9oa"
      >
        LABEL
      </h1>
    </a>
     
    <div
      class="svelte-1rse9oa"
    >
      <span
        class="subtitle svelte-1rse9oa"
      >
        SCHEMA LABEL
      </span>
       
      <div
        class="assignee svelte-1rse9oa"
      />
       
      <footer
        class="footer svelte-1rse9oa"
      >
        Last changed 
        1/1/2020 12:00 AM
      </footer>
    </div>
  </section>
</div>
`;
